File: src/main/kotlin/com/example/ioc_spring/modelo/ItemFactura.java
package com.example.ioc_spring.modelo;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ItemFactura {

    private Producto producto;
    private int cantidad;
    private double totalPagado;

    public ItemFactura() {
    }

    public ItemFactura(Producto producto, int cantidad) {
        this.producto = producto;
        this.cantidad = cantidad;
        this.totalPagado = calculoProducto();
    }

    private double calculoProducto() {
        double monto = this.producto.getPrecioProducto() * this.cantidad;
        return Math.round(monto * 100.0) / 100.0;
    }
}

File: src/main/kotlin/com/example/ioc_spring/modelo/ItemFactura.kt
package com.example.ioc_spring.modelo

import lombok.Getter
import lombok.Setter
import kotlin.math.roundToInt

@Getter
@Setter
class ItemFactura(producto: Producto?, cantidad: Int?) {

    var producto: Producto? = null
    var cantidad: Int? = null
    var totalPagado = 0.0

    init {
        this.producto = producto
        this.cantidad = cantidad
        totalPagado = calculoProducto()
    }

    private fun calculoProducto(): Double {
        var estado = 0.0
        val monto = this.producto?.precioProducto?.times(this.cantidad!!)
        if (monto != null) {
            estado = (monto * 100.0).roundToInt() / 100.0
        }
        return estado
    }
}